server/controllers/user.js

```diff
@@ -34,45 +49,58 @@ exports.updatePersonalInformation = asyncHandler(async (req, res, next) => {
       { headline: headline, bio: bio, location: location },
       { new: true },
     );
+    profile = getProfile(user);
     res.status(202).json({
       success: true,
-      user: {
-        email: user.get('email'),
-        username: user.get('username'),
-        headline: user.get('headline'),
-        bio: user.get('bio'),
-        location: user.get('location'),
-        profilePicUrl: user.get('profilePicUrl'),
-      },
+      profile,
     });
   } catch (err) {
     res.status(400);
     throw new Error('Failed to update personal information');
   }
 });
 
-exports.updateProfilePicture = asyncHandler(async (req, res, next) => {
-  const { url } = req.body;
+exports.getUserInfo = asyncHandler(async (req, res, next) => {
+  const username = req.params.username;
```

Comment:
We are using the auth middleware for this route. Maybe access the user id from the request object rather than using username as a path parameter? :)